### PR TITLE
refactor: update nova permission changes based on msq

### PR DESCRIPTION
**Pull request by alexbarnsley**
## Summary

https://app.clickup.com/t/1kjenrh

Related to https://github.com/ArkEcosystem/marketsquare.io/pull/2367 and used by https://github.com/PayvoHQ/payvo.com/pull/151

## Checklist

-   [ ] I checked my UI changes against the design and there are no notable differences
-   [ ] I checked my UI changes for any responsiveness issues
-   [x] I checked my (code) changes for obvious issues, debug statements and commented code
-   [ ] I provided a screenshot of my changes to the component _(if applicable)_
-   [ ] I regenerated the `icons.html` file and checked if my newly added icon is shown correctly _(if necessary)_
-   [ ] I added an explanation on how to use the component to the readme _(if necessary)_
-   [ ] Documentation _(if necessary)_
-   [ ] Tests _(if necessary)_
-   [x] Ready to be merged

### CODE REVIEW

File: src/Fortify/Console/Commands/RunPlaybookCommand.php [hunk 1]
```diff
@@ -104,14 +104,17 @@ private function askPlaybookName(array $availablePlaybooks): string
...
+         */
         $files = scandir(__DIR__.'/../Playbooks');
 
         unset($files[0], $files[1]);
 
         return array_map(fn (string $file) => str_replace('.php', '', $file), $files);
     }
 
-    private function resolvePlaybookDefinition($class): PlaybookDefinition
+    private function resolvePlaybookDefinition(mixed $class): PlaybookDefinition
```
Comment (leMaur) [inline src/Fortify/Console/Commands/RunPlaybookCommand.php:117]: @alexbarnsley I prefer using `mixed` as a last resort. What do you think if we use union type instead?

```suggestion
    /** @param PlaybookDefinition|Playbook|class-string $class */
    private function resolvePlaybookDefinition(PlaybookDefinition|Playbook|string $class): PlaybookDefinition
```

File: src/Fortify/Console/Commands/RunPlaybookCommand.php [hunk 2]
```diff
@@ -104,14 +104,17 @@ private function askPlaybookName(array $availablePlaybooks): string
 
     private function getAvailablePlaybooks(): array
     {
+        /**
+         * @var array
+         */
```
Comment (leMaur) [inline src/Fortify/Console/Commands/RunPlaybookCommand.php:109]:
```suggestion
        /** @var array<string> */
```

**Comment by leMaur** [comment]
@alexbarnsley if you want you can improve `RunPlaybookCommand` by adding doc-block to these methods too
```php
/**
 * @param array<class-string> $availablePlaybooks
 * @return class-string
 */
private function askPlaybookName(array $availablePlaybooks): string
```

```php
/** @return array<class-string> */
private function getAvailablePlaybooks(): array
```

**Comment by alexbarnsley** [comment]
@leMaur I can do, although this file was just copied over from this MSQ PR: https://github.com/ArkEcosystem/marketsquare.io/pull/2387 :thinking:

**Comment by alexbarnsley** [comment]
tbh I'm happy to leave it as is as it's only for playbooks which are mostly used on first-runs or during development. @ItsANameToo doth thou agree'th?

**Comment by leMaur** [comment]
> tbh I'm happy to leave it as is as it's only for playbooks which are mostly used on first-runs or during development. @ItsANameToo doth thou agree'th?

@alexbarnsley think you're right...maybe it's too much 😅

**Comment by ItsANameToo** [comment]
I'm inclined to go with leaving it as `mixed` for the parameter as it's only playbooks and that makes it a lot more readable while we don't gain much from having the specific types listed there as it's set and forget for those methods.

**Review by ItsANameToo — APPROVED** [review]

**Pull request merged**